plugins/obs-ffmpeg/obs-ffmpeg-source.c

```diff
@@ -237,13 +248,14 @@ static void dump_source_info(struct ffmpeg_source *s, const char *input,
 		"\tis_hw_decoding:          %s\n"
 		"\tis_clear_on_media_end:   %s\n"
 		"\trestart_on_activate:     %s\n"
+		"\tffmpeg_options:          %s\n"
 		"\tclose_when_inactive:     %s",
 		input ? input : "(null)",
 		input_format ? input_format : "(null)", s->speed_percent,
 		s->is_looping ? "yes" : "no", s->is_linear_alpha ? "yes" : "no",
 		s->is_hw_decoding ? "yes" : "no",
 		s->is_clear_on_media_end ? "yes" : "no",
-		s->restart_on_activate ? "yes" : "no",
+		s->restart_on_activate ? "yes" : "no", s->ffmpeg_options,
```

Comment:
I think the "ffmpeg_options" should be last in the logged output.

Reply:
thanks for picking up on that search/replace miss.  I've move the order of log.

if I split this into two PRs then one wont compile without the other being pulled in first.

I'm happy to detail as much as possible in the one PR to make it obvious but if you want to give me the text that you'll be happy with then it will avoid a ping-pong messages

Reply:
Not two PRs, two _commits_.  Put two commits in this PR.

Reply:
sorry, i need to read things properly

before I pull it in here, is this what you would like to see?

https://github.com/stopforumspam/obs-studio/commits/ffmpeg_dev

Reply:
No.  You want deps/media-playback commit _first_, then the obs-ffmpeg commit second.

> Technically, this PR should actually be two commits: the first one for the "deps/media-playback" changes, and the second one for the "obs-ffmpeg" changes.

In the order you've put them, the first commit will probably fail to compile because the deps/media-playback changes don't exist.  Even if it didn't, the obs-ffmpeg commit on its own being first would do nothing.  You want commits in a logical order where each commit can compile on its own.

The commit prefix for deps/media-playback is "deps/media-playback".  The commit prefix for obs-ffmpeg is "obs-ffmpeg".  The deps/media-playback commit should not try to include an extra "UI: " prefix in its commit message, because it doesn't touch the UI files.  Commit prefixes are based on what files they modify.  If you want to check what the appropriate commit prefix should be, check the commit history for that section/folder of code.

The deps/media-playback commit is not accurate.
> Adds a text entry field to the Media Source for non-local files
> 
> This list of options is parsed by the obs-ffmpeg plugin to control FFmpeg

It would be more accurate to say that it "Adds support for setting FFmpeg options for media playback".  The obs-ffmpeg plugin doesn't do any parsing either, it passes the property to deps/media-playback, which parses it, and passes it to FFmpeg.

I'm not sure that the obs-ffmpeg commit message is accurate either.  There isn't always a list of options specified.  A more accurate commit message body would be "Provides a user option to specify FFmpeg options for a media source."

Reply:
All none.  thanks for the guidance.  If I decide to jump into more code changes in the future then I'll be sure to revisit this PR